firedrake/slate/slac/compiler.py

```diff
@@ -290,13 +583,17 @@ def eigen_matrixbase_type(shape):
              library syntax.
     """
     if len(shape) == 0:
-        raise NotImplementedError("Scalar-valued expressions cannot be declared as an Eigen::MatrixBase object.")
+        raise NotImplementedError(
+            "Scalar exprs cannot be declared as an Eigen::MatrixBase object."
```

Comment:
@wence- I finally solved this problem. I still do the `Eigen::MatrixBase<double, 1, 1>` hack, but @miklos1 pointed out some useful information I needed to know at the PyOP2 level (specifically for assembling 0-forms). Next commit will add zero-form tests in Slate and should be ready for review.